src/main/scala/com/codacy/client/bitbucket/v2/service/RepositoryServices.scala

```diff
@@ -10,10 +10,21 @@ class RepositoryServices(client: BitbucketClient) {
    * Gets the list of the user's repositories. Private repositories only appear on this list
    * if the caller is authenticated and is authorized to view the repository.
    */
-  def getRepositories(ownerInfo: OwnerInfo, pageLength: Option[Int] = Option(100)): RequestResponse[Seq[Repository]] = {
-    val pagelen = pageLength.fold("")(pagelen => s"?pagelen=$pagelen")
+  def getRepositories(
+      ownerInfo: OwnerInfo,
+      pageLength: Option[Int] = Option(100),
+      userRole: Option[Role] = None
+  ): RequestResponse[Seq[Repository]] = {
+    val queryParameters = pageLength.fold {
```

Comment:
I would prefer a "add parameters method" like:

  def joinQueryString(baseUrl: String, queryString: String): String = {
    val parsedQuery = queryString.stripPrefix("&").stripPrefix("?")
    if (parsedQuery.nonEmpty) {
      baseUrl match {
        case url if url.contains('?') =>
          baseUrl + "&" + parsedQuery
        case url =>
          baseUrl + "?" + parsedQuery
      }
    } else {
      baseUrl
    }
  }

Reply:
Since there is no easy way to do this, lets move the code as @machadoit  suggested to a helper url and use that.